core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -280,6 +282,11 @@ public int onDataRead(
                 logInvalidStream = true;
             }
         } else {
+            if (req.abortedResponse()) {
+                // Additional DATA frame can be sent after the response has been aborted.
+                return dataLength + padding;
```

Comment:
Maybe you meant this:
```suggestion
                // Discard the DATA frame received after the response has been aborted
                // because an aborted response means we have finished handling the
                // request.
                return dataLength + padding;
```